### PR TITLE
Item page improvements

### PR DESCRIPTION
**Pull request by ThibaultNocchi**
I'm trying to improve the background code of the item page.

* Factorized the track selector into their own component, which emits an `input` event when the track is changed with the index of the track. It also chooses the default value, disabled the selector if necessary, applies placeholders... 
* Factorized the `cols` `sm` and `class` tags of the two-columns lines in the info part (genres, writing, selectors...) to easily edit them later to find the right sizing 🤌
* Warning message instead of the selectors when no media source are found
* Fix the play button not being centered anymore on mobile size

### CODE REVIEW

**Comment by ferferga** [comment]
Take in mind that backdrop store doesn't work in master, is fixed in #520

**Comment by ThibaultNocchi** [comment]
Waiting for #530 as it was rebased against

File: components/Item/TrackSelector.vue [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    item: {
+      type: Object as PropType<BaseItemDto>,
+      required: true
+    },
+    // Current media source used (current movie version for instance)
+    mediaSourceIndex: {
+      type: Number,
+      default: 0
+    },
+    // Which media type to consider for this selector
```
Comment (ferferga) [inline components/Item/TrackSelector.vue:63]: We should use jsdoc syntax for these as well imo

File: components/Item/TrackSelector.vue [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  },
+  watch: {
+    /**
+     * @param {number} newVal - New index value choosen in the v-select
+     */
+    trackIndex(newVal: number): void {
+      this.$emit('input', newVal);
+    },
+    /**
+     * When the media source index is changed by the parent, we reset the selected track as they have changed
```
Comment (ferferga) [inline components/Item/TrackSelector.vue:168]:
```suggestion
     * When the media source index is changed by the parent, we reset the selected track as it has changed
```

File: pages/item/_itemId/index.vue [hunk 3]
```diff
@@ -155,29 +152,19 @@
...
-                v-if="
-                  item &&
-                  ((item.MediaSources && item.MediaSources.length > 1) ||
-                    videoTracks.length > 0 ||
-                    audioTracks.length > 0 ||
-                    subtitleTracks.length > 0)
-                "
-                class="mt-2"
-              >
+              <div v-if="item && item.MediaSources" class="mt-2">
```
Comment (ferferga) [inline pages/item/_itemId/index.vue:155]: Why the other conditions were removed?

File: pages/item/_itemId/index.vue [hunk 4]
```diff
@@ -198,136 +185,64 @@
                     </v-select>
                   </v-col>
                 </v-row>
-                <v-row v-if="videoTracks.length > 0" align="center">
+                <v-row align="center">
```
Comment (ferferga) [inline pages/item/_itemId/index.vue:188]: Same

**Review by ferferga — CHANGES_REQUESTED** [review]

File: pages/item/_itemId/index.vue (hunk 3 above)
Comment (ThibaultNocchi) [reply pages/item/_itemId/index.vue:155]: I wanted to show the different selectors even when there was nothing to pick, as having "No video/audio/sub available" is better than getting nothing at conveying the fact that the media has no tracks.

Comment (ThibaultNocchi) [reply pages/item/_itemId/index.vue:155]: Though I should add a placeholder in the media source vselect

Comment (ferferga) [reply pages/item/_itemId/index.vue:155]: And disable it in that case, so the "No available X" is always shown but interaction with the v-select is completely disabled.

File: pages/item/_itemId/index.vue [hunk 5]
```diff
@@ -380,21 +304,27 @@
...
+interface TwoColsInfoColumn {
+  lCols: number;
+  lSm: number;
+  rCols: number;
+  rSm: number;
+  lClass: { [key: string]: boolean };
+}
 
 export default Vue.extend({
+  components: { TrackSelector },
```
Comment (ferferga) [inline pages/item/_itemId/index.vue:327]: This is not needed as we have all the components autoimported

File: pages/item/_itemId/index.vue [hunk 6]
```diff
@@ -198,139 +188,73 @@
...
+                  <v-col
+                    :cols="twoColsInfoColumn.rCols"
+                    :sm="twoColsInfoColumn.rSm"
+                  >
+                    <track-selector
+                      :item="item"
+                      :media-source-index="currentSourceIndex"
+                      :type="'Subtitle'"
+                      @input="currentSubtitleTrack = $event"
+                    ></track-selector>
```
Comment (ferferga) [inline pages/item/_itemId/index.vue:248]:
```suggestion
                    />
```

File: pages/item/_itemId/index.vue [hunk 7]
```diff
@@ -198,139 +188,73 @@
...
+                  <v-col
+                    :cols="twoColsInfoColumn.rCols"
+                    :sm="twoColsInfoColumn.rSm"
+                  >
+                    <track-selector
+                      :item="item"
+                      :media-source-index="currentSourceIndex"
+                      :type="'Audio'"
+                      @input="currentAudioTrack = $event"
+                    ></track-selector>
```
Comment (ferferga) [inline pages/item/_itemId/index.vue:228]:
```suggestion
                    />
```

File: pages/item/_itemId/index.vue [hunk 8]
```diff
@@ -198,139 +188,73 @@
...
+                  <v-col
+                    :cols="twoColsInfoColumn.rCols"
+                    :sm="twoColsInfoColumn.rSm"
+                  >
+                    <track-selector
+                      :item="item"
+                      :media-source-index="currentSourceIndex"
+                      :type="'Video'"
+                      @input="currentVideoTrack = $event"
+                    ></track-selector>
```
Comment (ferferga) [inline pages/item/_itemId/index.vue:208]:
```suggestion
                    />
```

**Review by ferferga — CHANGES_REQUESTED** [review]

File: pages/item/_itemId/index.vue (hunk 5 above)
Comment (ThibaultNocchi) [reply pages/item/_itemId/index.vue]: Done

File: pages/item/_itemId/index.vue (hunk 6 above)
Comment (ThibaultNocchi) [reply pages/item/_itemId/index.vue]: Done

File: pages/item/_itemId/index.vue (hunk 7 above)
Comment (ThibaultNocchi) [reply pages/item/_itemId/index.vue]: Done

File: pages/item/_itemId/index.vue (hunk 8 above)
Comment (ThibaultNocchi) [reply pages/item/_itemId/index.vue]: Done

File: pages/item/_itemId/index.vue [hunk 9]
```diff
@@ -42,7 +42,7 @@
             </div>
             <v-row
               class="mt-4 align-center"
-              :class="{ 'text-center': !$vuetify.breakpoint.mdAndUp }"
+              :class="{ 'justify-center': !$vuetify.breakpoint.mdAndUp }"
             >
               <v-btn
                 v-if="canPlay(item)"
```
Comment (ThibaultNocchi) [inline pages/item/_itemId/index.vue:48]: Here we're using a v-if which calls to the mixin and a disabled which calls to local method, do these overlap? @MrTimscampi

Comment (ferferga) [reply pages/item/_itemId/index.vue:48]: @ThibaultNocchi they're combined, see audio controls for an example

Comment (ferferga) [reply pages/item/_itemId/index.vue:48]: You can remove the braces though

**Comment by ThibaultNocchi** [comment]
I've removed the icon and now the play button gets disabled when there's no media source
Screenshot_2021-01-16 Jellyfin(1)

**Review by ferferga — APPROVED** [review]

**Pull request merged**